### PR TITLE
EOS-18876: Added query parameter support for getting s3_urls from backend for supported schemas - http,https,s3

### PR DESCRIPTION
**Pull request by rohitkolapkar**
# Backend

## Problem Statement
<pre>
Story Ref (if any):
</pre>
[EOS-18876](https://jts.seagate.com/browse/EOS-18876)
## Unit testing on RPM done
<pre>
Yes
</pre>
## Problem Description
<pre>
On backend, there is a dedicated endpoint ({{GET /api/v1/s3) }}to get s3 URLs. 
But it is not returning the s3 url in the required format s3://ip-address:port. 
</pre>
## Solution
<pre>
Added query parameter support for getting s3_urls from backend for supported schemas - http,https,s3
After merge of this PR, csm_web code should be modified and consume the GET api -
 /api/{version}/s3?schemas=s3 to get the s3_url and add it to the payload of Lyve Pilot registration api.
</pre>
## Unit Test Cases
<pre>

</pre>
image
image
image
image
Signed-off-by:

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

**Comment by tb-s** [comment]
@rohitkolapkar , should commit 5cb0dc8 be part of this PR as well?  If not, we should remove it.

File: csm/core/services/s3/server_info.py [hunk 1]
```diff
@@ -15,18 +15,26 @@
...
 
 class S3ServerInfoService(ApplicationService):
     """
     Service for acessing S3 server information
     """
 
-    async def get_s3_server_info(self):
+    async def get_s3_server_info(self, schemas):
         """
         Obtain information about S3 server in json format
```
Comment (tb-s) [inline csm/core/services/s3/server_info.py:28]: We forgot to add documentation to the new `schemas` parameter here.  Also, let's add documentation for the return value as well.

File: csm/core/services/s3/server_info.py [hunk 2]
```diff
@@ -15,18 +15,26 @@
...
         """
-        s3_urls = ServiceUrls.get_s3_uris()
+        supported_schemas = ServiceUrls.get_s3_supported_schemas()
+        if schemas is not None:
+            if not all(s in supported_schemas for s in schemas):
+                raise InvalidRequest(f'Unsupported schema in list of schemas {schemas}. '
+                    f'Supported schemas are {supported_schemas}')
+            s3_urls = [ServiceUrls.get_s3_uri(s) for s in schemas]
+        else:
+            s3_urls = [ServiceUrls.get_s3_uri(s) for s in supported_schemas]
```
Comment (tb-s) [inline csm/core/services/s3/server_info.py:37]: I have just realized that a minor rewrite here to get rid of the very similar `s3_urls = [ServiceUrls.get_s3_uri(s) ...` lines and move them out of the conditional is possible.  This is not a required change, though.

**Review by tb-s — CHANGES_REQUESTED** [review]

File: csm/core/services/s3/server_info.py (hunk 1 above)
Comment (rohitkolapkar) [reply csm/core/services/s3/server_info.py:28]: @tb-s Added documentation. please check.

Comment (tb-s) [reply csm/core/services/s3/server_info.py:28]: I believe you added documentation to another function.  Also, I have just noticed that this function does not need to be `async` anymore.  We can remove it and adapt its calls.

**Review by tb-s — APPROVED** [review]

**Review by kupranay — APPROVED** [review]

**Review by kupranay — APPROVED** [review]

**Review by soniyamoholkar — APPROVED** [review]

**Review by avrnv — APPROVED** [review]

**Pull request merged**